core/src/Pos/Core/NetworkMagic.hs

```diff
@@ -0,0 +1,47 @@
+module Pos.Core.NetworkMagic
+       ( NetworkMagic (..)
+       , makeNetworkMagic
+       ) where
+
+import           Universum
+
+import           Data.SafeCopy (SafeCopy (..), contain, safeGet, safePut)
+import           Data.Serialize (getWord8, putWord8)
+import qualified Data.Text.Buildable as Buildable
+import           Formatting (bprint, build, (%))
+
+import           Pos.Crypto.Configuration (ProtocolMagic (..), RequiresNetworkMagic (..),
+                                           getProtocolMagic)
+import           Pos.Util.Util (cerealError)
+
+
+--------------------------------------------------------------------------------
+-- NetworkMagic
+--------------------------------------------------------------------------------
+
+-- mhueschen: although it makes no sense for an identifier to have a sign, we're
+-- opting to maintain consistency with `ProtocolMagicId`, rather than risk subtle
+-- conversion bugs.
+data NetworkMagic
```

Comment:
I really don't like this type name. What is `Magic`? What does it mean, what does it do, why do we care about it?

Reply:
Open to modifying this. `NetworkMagic` was a suggestion from @dcoutts which we ran with.

Reply:
No. `NetworkMagic` is probably the best name for it.